File: angular/projects/admin/src/app/data/skill-groups/skill-group.ts
import { Skill } from "../skills/skill";

export interface SkillGroup{
    title: string,
    displayNumber: number,
    id: number,
    skills: Skill[]
}

export class ListSkillGroup implements SkillGroup {
    title: string;
    displayNumber: number;
    id: number;
    skills: Skill[];
    
    inEditMode: boolean = false;

    constructor(skillGroup: SkillGroup) {
        this.title = skillGroup.title;
        this.displayNumber = skillGroup.displayNumber;
        this.id = skillGroup.id;
        this.skills = skillGroup.skills;
    }
}

File: api/src/Portfolio.Domain/Models/SkillGroup.cs
using Portfolio.Domain.Models.Common;
using System.Collections.Generic;

namespace Portfolio.Domain.Models;

public class SkillGroup : BaseEntity, IHasDisplayNumber
{
    #region Properties

    public string Title { get; set; }

    public int DisplayNumber { get; set; }

    public IEnumerable<Skill> Skills { get; set; }

    #endregion
}
